### ISSUE TITLE
bug: CapacitorHttp does not work with Angular withFetch() interceptor

### ISSUE THREAD

**Issue by muuvmuuv** [ionic-team/capacitor#7145]
### Capacitor Version

```
Latest Dependencies:

  @capacitor/cli: 5.6.0
  @capacitor/core: 5.6.0
  @capacitor/android: 5.6.0
  @capacitor/ios: 5.6.0

Installed Dependencies:

  @capacitor/cli: 5.5.1
  @capacitor/android: 5.5.1
  @capacitor/core: 5.5.1
  @capacitor/ios: 5.5.1
```

### Platform(s)

- Android
- iOS

### Current Behavior

When using withFetch from Angular 17, we can no longer use CapacitorHttp POST requests on Android, it does not proceed the request on native side but passes it without any more headers to native window.fetch.

### Code Reproduction

https://github.com/muuvmuuv/ionic-issue-7145

### Other Technical Details

`pnpm --version` output: 8.10.3

`node --version` output: v20.9.0

`pod --version` output (iOS issues only): 1.14.2

**Comment by muuvmuuv** [comment]
I have added a repro but it does not reproduce my error atm, I will continue investigating and update the ticket as soon as I have more information.

I can just confirm that removing withFetch it in our production app fixes it. My guess is that the patching of fetch has some issues.

**Comment by jcesarmobile** [comment]
Your reproduction uses 5.6.0, while in your ticket you said you were in 5.5.1, so maybe it was fixed? there were several http fixes in 5.6.0. Can you try updating your app to 5.6.0?

**Comment by muuvmuuv** [comment]
True, maybe a reason why my repl did not work. Lets close this for now. I am on vacation till January. If I will encounter it again, I will re-open.
